pkg/webhook/utils/resources.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2020-2021 Clastix Labs
+// SPDX-License-Identifier: Apache-2.0
+
+package utils
+
+import (
+	"context"
+
+	networkingv1 "k8s.io/api/networking/v1"
+	networkingv1beta1 "k8s.io/api/networking/v1beta1"
+	schedulev1 "k8s.io/api/scheduling/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const TRUE string = "true"
+
+// Get PriorityClass by name (Does not return error if not found).
+func GetPriorityClassByName(ctx context.Context, c client.Client, name string) (*schedulev1.PriorityClass, error) {
+	class := &schedulev1.PriorityClass{}
+	if err := c.Get(ctx, types.NamespacedName{Name: name}, class); err != nil {
+		return nil, err
+	}
+
+	return class, nil
+}
+
+// Check if given PriorityClass is cluster default.
+func IsDefaultPriorityClass(class *schedulev1.PriorityClass) bool {
+	if class != nil {
+		if class.GlobalDefault {
+			return true
+		}
+	}
+
+	return false
+}
+
+// Get StorageClass by name (Does not return error if not found).
+func GetStorageClassByName(ctx context.Context, c client.Client, name string) (*storagev1.StorageClass, error) {
+	class := &storagev1.StorageClass{}
+	if err := c.Get(ctx, types.NamespacedName{Name: name}, class); err != nil {
+		return nil, err
+	}
+
+	return class, nil
+}
+
+// Check if given StorageClass is cluster default.
+func IsDefaultStorageClass(class client.Object) bool {
+	annotation := "storageclass.kubernetes.io/is-default-class"
+
+	if class != nil {
+		for a, v := range class.GetAnnotations() {
+			if a == annotation && v == TRUE {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+// Get IngressClass by name (Does not return error if not found).
+func GetIngressClassByName(ctx context.Context, c client.Client, ingressClassName *string) (client.Object, error) {
+	if ingressClassName == nil {
+		return nil, nil
+	}
+
+	var obj client.Object
+
+	version, _ := GetK8sVersion()
```

Comment:
We're going to issue the Kubernetes version retrieval for each request: since this function is used in the file `pkg/webhook/defaults/handler.go`, please, initialize the retrieved version when calling `Handler`, considering that in `main.go` line 230 we already have the `kubeVersion` object, so you could inject it as a parameter.